hack/verify-archives.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About:
+# This script extracts the golden image tarballs and compares their contents against the
+# already-tracked golden folder (which is used by Bazel to generate deterministic images
+# on every e2e test invocation). The purpose of this script is to verify that the tarballs
+# residing in golden-archives matches the original source of truth (golden).
+#
+# Usage:
+#   verify-archives.sh repo-root
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "$#" != 1 ]]
+then
+    echo "Error: required argument missing!" >&2
+    echo "Usage verify-archives.sh repo-root" >&2
+    exit 1
+fi
+
+repoRoot=$1
+paths=(
+    bar/1.0
+    foo/1.0-linux_amd64
+    foo/1.0-linux_s390x
+    foo/NOTAG-0
+)
+# setup mock directory structure
+mockGolden=$(mktemp -d)
+mkdir "${mockGolden}/foo"
+mkdir "${mockGolden}/bar"
+
+goldenArchive="${repoRoot}/test-e2e/cip/golden-archives/"
+goldenContent="${repoRoot}/test-e2e/cip/golden"
+
+# looping through image archives
+for path in "${paths[@]}"
+do
+    # get parent directory and file name from path
+    goldenFile="$(cut -d "/" -f2 <<< "$path")"
+    goldenParent="$(cut -d "/" -f1 <<< "$path")"
+    # load image from tarball archive
+    output=$(docker load -i "${goldenArchive}${path}.tar")
+    # reformat docker load output to only grab the image ID
+    img=$(echo "$output" | awk '/sha256:/ {print $4}' | cut -d ":" -f2)
+    # save container ID after creation
+    id=$(docker create "$img" /dev/null)
+    # specify the data file's parent directory within the container
+    containerParent=$(echo "$path" | sed -e 's/^[^\/]*//g' -e 's/1.0-//g' -e 's/-0//g' -e 's/\/1.0//g')
```

Comment:
The container directory structure is a bit maddening. Here's what I mean:
```
Img                          Path-to-data-file-inside-container
bar/1.0                      /golden/bar/1.0
foo/1.0-linux_amd64          /golden/foo/linux_amd64/1.0-linux_amd64
foo/1.0-linux_s390x          /golden/foo/linux_s390x/1.0-linux_s390x
foo/NOTAG-0                  /golden/foo/NOTAG/NOTAG-0
```
Notice the inconstancy in parent directories? The names even change!
This is the reason for such a long string of sed commands.

Reply:
Hmm, instead of doing a long series of sed calls (which hurts readability), you could use a bash associative array to assign keys/values. That should simplify this area.